src/Core/src/Response.php

```diff
@@ -85,24 +86,74 @@ public function resolve(?float $timeout = null): bool
                 }
             }
 
-            $statusCode = $this->httpResponse->getStatusCode();
+            return $this->handleStatus();
         } catch (TransportExceptionInterface $e) {
             throw $this->resolveResult = new NetworkException('Could not contact remote server.', 0, $e);
         }
+    }
 
-        if (500 <= $statusCode) {
-            throw $this->resolveResult = new ServerException($this->httpResponse);
-        }
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $responses
+     * @param float|null $timeout   Duration in seconds before aborting. When null wait
+     *                              until the end of execution. Using 0 means non-blocking
+     *
+     * @return iterable<self>
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function multiplex(iterable $responses, float $timeout = null): iterable
+    {
+        /** @var self[] $responseMap */
+        $responseMap = [];
+        $httpResponses = [];
+        $httpClient = null;
+        foreach ($responses as $response) {
+            if (null !== $response->resolveResult) {
+                yield $response;
+
+                continue;
+            }
 
-        if (400 <= $statusCode) {
-            throw $this->resolveResult = new ClientException($this->httpResponse);
+            if (null === $httpClient) {
+                $httpClient = $response->httpClient;
+            }
+            $httpResponses[] = $response->httpResponse;
+            $responseMap[\spl_object_id($response->httpResponse)] = $response;
         }
 
-        if (300 <= $statusCode) {
-            throw $this->resolveResult = new RedirectionException($this->httpResponse);
+        // response is empty
+        if (null === $httpClient) {
+            return;
```

Comment:
Throw an error here instead. 

And at the very top of the function do a check `if (empty($responses)) { return;}`

Reply:
thanks